### PR TITLE
feat: Add includeScopes query param to credentials, workflows, and projects

### PR DESCRIPTION
**Pull request by valya**
## Summary
Add `includeScopes` query parameter to return the scopes that the user had on specific credentials, workflows, and projects.

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [x] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/cli/src/services/role.service.ts [hunk 1]
```diff
@@ -132,4 +144,68 @@ export class RoleService {
...
+			return entity;
+		}
+
+		if (!('active' in entity) && !('type' in entity)) {
+			throw new ApplicationError('Cannot detect if entity is a workflow or credential.');
+		}
+
+		const globalScopes = this.getRoleScopes(user.role).filter((s) =>
+			s.startsWith('active' in entity ? 'workflow:' : 'credential:'),
+		);
```
Comment (krynble) [inline packages/cli/src/services/role.service.ts:186]: I've noticed a difference here, when adding scopes to workflows and credentials, we check only for scopes that `startsWith` but when fetching for projects, we just add all global scopes. Should we perhaps do the same for projects?

Comment (krynble) [reply packages/cli/src/services/role.service.ts:186]: I just tried changing the code to accomodate this comment and noticed that the project-specific scopes contain `workflow:*` and `credential:*` scopes, so they span over more than just `project:*` hence it makes senes to return everything here. I wonder whether we can optimiza this in the future, but it should be fine for now.

**Review by krynble — APPROVED** [review]

**Review by krynble — APPROVED** [review]

**Review by krynble — COMMENTED** [review]
One thing I noticed to be missing is the option to provide a `projectId´ for credentials to return only the credentials that can be used within that project.

**Comment by valya** [comment]
> One thing I noticed to be missing is the option to provide a `projectId´ for credentials to return only the credentials that can be used within that project.

There's already one, it's part of the credentials filter (`credentials.filter.dto.ts`).

**Pull request merged**